packages/ingest/src/embedding/handleEmbeddingBatchSuccess.ts

```diff
@@ -0,0 +1,96 @@
+import { PrismaClientWithAccelerate } from "@oakai/db";
+
+import { IngestError } from "../IngestError";
+import { updateLessonsState } from "../db-helpers/updateLessonsState";
+import { downloadOpenAiFile } from "../openai-batches/downloadOpenAiFile";
+import { jsonlToArray } from "../utils/jsonlToArray";
+import { parseBatchEmbedding } from "./parseBatchEmbedding";
+
+/**
+ * When an embedding batch is successfully processed, this function
+ * extracts the embeddings and updates the lesson plan parts with them.
+ */
+export async function handleEmbeddingBatchSuccess({
+  prisma,
+  ingestId,
+  batchId,
+  outputFileId,
+}: {
+  prisma: PrismaClientWithAccelerate;
+  ingestId: string;
+  batchId: string;
+  outputFileId: string;
+}) {
+  const { file } = await downloadOpenAiFile({
+    fileId: outputFileId,
+  });
+  const text = await file.text();
+  const jsonArray = jsonlToArray(text);
+
+  const lessonIdsFailed: Set<string> = new Set();
+  const lessonIdsCompleted: Set<string> = new Set();
+
+  for (const json of jsonArray) {
+    let lessonId: string | undefined = undefined;
+    try {
+      const batchEmbedding = parseBatchEmbedding(json);
+      lessonId = batchEmbedding.lessonId;
+      console.log(`Embedding lesson ${lessonId}`);
+      const { lessonPlanPartId, embedding } = batchEmbedding;
+
+      const vector = `[${embedding.join(",")}]`;
+      const res = await prisma.$executeRaw`
+              UPDATE ingest.ingest_lesson_plan_part
+              SET embedding = ${vector}::vector 
+              WHERE id = ${lessonPlanPartId}`;
+
+      if (res !== 1) {
+        lessonIdsFailed.add(lessonId);
+        continue;
+      }
+    } catch (error) {
+      if (error instanceof IngestError && error.lessonId) {
+        lessonId = error.lessonId;
+      }
+      if (lessonId) {
+        lessonIdsFailed.add(lessonId);
+      }
```

Comment:
Do we want to track the error here? Maybe push to an array of errors to log at the end?

Reply:
Thanks yes was def thinking of this -- glad you brought it up. Will make this change before merging